broker-core/src/main/java/io/zeebe/broker/clustering/base/partitions/PartitionLeaderElection.java

```diff
@@ -68,23 +92,150 @@ public void start(ServiceStartContext startContext) {
         e -> {
           election = e;
           election.run(memberId);
+          startContext.getScheduler().submitActor(this).join();
           startFuture.complete(null);
         });
 
     startContext.async(startFuture, true);
   }
 
+  @Override
+  protected void onActorStarted() {
+    initListeners();
+  }
+
+  private void initListeners() {
+    election.addListener(this);
+    election.addStateChangeListener(this);
+    try {
+      final Leader<String> currentLeader = election.getLeadership().leader();
+      if (memberId.equals(currentLeader.id())) {
+        transitionToLeader(currentLeader.term());
+      } else {
+        transitionToFollower();
+      }
+    } catch (PrimitiveException e) {
+      LOG.error(
+          "Couldn't get current leadership for partition {}. Transitioning to follower. {}",
```

Comment:
```suggestion
          "Couldn't get current leadership for partition {}. Transitioning to follower.",
```

The exception doesn't need a parameter in the log statement. Just add it as the last argument.